### PR TITLE
Parse connection string for options object

### PR DESCRIPTION
**Pull request by southpolesteve**

### CODE REVIEW

File: test/unit/helper.spec.ts [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
       const err = {};
       const result = isResourceValid({ id: 1 }, err);
 
       assert.equal(result, false);
       assert.deepEqual(err, { message: "Id must be a string." });
-      done();
+    });
+  });
+  describe("parseConnectionString", function() {
+    it("parses a valid connection string", function() {
```
Comment (christopheranderson) [inline test/unit/helper.spec.ts]: Please add case for how the error message looks/etc. when it is an invalid connection string.

File: src/CosmosClient.ts [hunk 2]
```diff
@@ -51,7 +52,11 @@ export class CosmosClient {
    */
 
   private clientContext: ClientContext;
-  constructor(private options: CosmosClientOptions) {
+  constructor(options: string | CosmosClientOptions) {
```
Comment (christopheranderson) [inline src/CosmosClient.ts]: I know it's annoying, but can we do this with an overload instead?

```typescript
constructor(connectionString: string)
constructor(options: CosmosClientOptions)
constructor(optionsOrConnectionString: string | CosmosClientOptions)
```

Even if you don't want to overload it, you should at least use the optionsOrConnectionString name (but I like overloading :))

Comment (christopheranderson) [reply src/CosmosClient.ts]: Also, need to update the JSdocs

File: test/functional/client.spec.ts [hunk 3]
```diff
@@ -31,6 +31,10 @@ describe("NodeJS CRUD Tests", function() {
       });
       assert.ok(client !== undefined, "client shouldn't be undefined if it succeeded");
     });
+    it("Accepts a connection string", function() {
+      const client = new CosmosClient(`AccountEndpoint=${endpoint};AccountKey=${masterKey};`);
```
Comment (christopheranderson) [inline test/functional/client.spec.ts]: error case

File: src/CosmosClient.ts (hunk 2 above)
Comment (southpolesteve) [reply src/CosmosClient.ts]: Done. Added overload and docs

**Pull request merged**